### PR TITLE
#3821 Button states CSS fixes

### PR DESCRIPTION
**Pull request by drakkein**
This PR fixes styling for hover and focussed states in our buttons styles. 

Changes could be verified via Storybook.

_Button primary_
image

_Button outline (secondary)_
image

_Button ghost (link)_
image

Also added new extra small size for all variations of buttons
image

And changed styling for circle button
image

### CODE REVIEW

File: packages/design-system/src/components/buttons/Button.module.scss [hunk 1]
```diff
@@ -19,21 +21,21 @@
     color: $neu-gull-gray;
   }
 
+  &:focus {
```
Comment (desfero) [inline packages/design-system/src/components/buttons/Button.module.scss]: let's not use `:focus` given we have already a polyfill for `focus-visible` included. You can check how focus was managed previously to see the differences

File: packages/design-system/.storybook/config.js [hunk 2]
```diff
@@ -3,6 +3,8 @@ import { addParameters, addDecorator } from "@storybook/react";
 import { withA11y } from "@storybook/addon-a11y";
 import { addReadme } from "storybook-readme";
 
+import "focus-visible";
```
Comment (desfero) [inline packages/design-system/.storybook/config.js]: Given that `focus-visible` is only used inside `design-system` package maybe we should do a `polyfills` file here and import it in the `web` app polyfills? If you think it's reasonable then we also need to add `sideEffects` property in package.json `  "sideEffects": [
    "*.sideEffect.*",
    "*.scss",
    "*.css"
  ],` for the design-system package and check if the polyfill is apllyied properly for the production build

File: packages/web/webpack/webpack.config.ci.prod.js [hunk 3]
```diff
@@ -22,6 +22,7 @@ module.exports = merge.smart(configCommon, {
     },
     runtimeChunk: true,
     minimize: false,
+    sideEffects: false,
```
Comment (drakkein) [inline packages/web/webpack/webpack.config.ci.prod.js]: hmmm.. somehow our `focus-visible` polyfill is being removed after optimization

**Comment by Moejoe90** [comment]
Blocked for now until https://github.com/Neufund/platform-frontend/pull/4133/

File: packages/web/webpack/webpack.config.ci.prod.js (hunk 3 above)
Comment (arjunu) [reply packages/web/webpack/webpack.config.ci.prod.js]: removed

**Pull request merged**